integration/test_hnsw_allow_replace_deleted.py

```
@@ -184,3 +185,125 @@ def test_index_add_after_rdb_load_with_deleted_elements(self):
+            f"Expected 1 search result, got {search_result[0]}"
+        assert search_result[1] == b"doc:1", \
+            f"Expected doc:1 to survive, got {search_result[1]}"
+        returned_fields = search_result[2]
+        vector_value = returned_fields[returned_fields.index(b"vector") + 1]
+        assert vector_value == vec1_new, \
+            f"Expected doc:1 to carry the updated vector, got {vector_value!r}"
+
+        # Deleting the survivor must actually remove it, proving label_lookup_
+        # resolves the label to the live slot rather than a tombstone.
+        client.delete("doc:1")
+        hnsw_index.info(client).num_docs == 0
```

Ruff (0.16.1)

[warning] Pointless comparison. Did you mean to assign a value? Otherwise, prepend `assert` or remove it.

(B015)